server/methods/core/shop.js

```diff
@@ -10,6 +10,62 @@ import { Reaction } from "/lib/api";
 import * as Collections from "/lib/collections";
 import * as Schemas from "/lib/collections/schemas";
 
+/**
+ * @name shop/updateBrandAsset
+ * @method
+ * @memberof Methods/Shop
+ * @param {Object} asset - brand asset {mediaId: "", type, ""}
+ * @param {String} shopId - the shop id coresponding to the shop for which
+ *                 the asset should be applied
+ * @return {Int} returns update result
+ */
+export updateShopBrandAssets(asset, shopId) {
+  check(asset, {
+    mediaId: String,
+    type: String
+  });
+  check(shopId, String);
+
+  // must have core permissions
+  if (!Reaction.hasPermission("core", null, shopId)) {
```

Comment:
added `shopId` to the permissions check.
I don't know the permissions code well, but I read the code and this feels right.

Reply:
the rest of this method is unchanged from the previous